### PR TITLE
Rename "gestures" store state to "actions"

### PR DESCRIPTION
**Pull request by microbit-grace**
Migrate local storage action data from "gestures" key to "actions" key

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -858,21 +827,35 @@ const createMlStore = (logging: Logging) => {
...
+          migrate(persistedStateUnknown, version) {
+            if (version === 0) {
+              // Rename the "gestures" field to "action"
+              const persistedState = persistedStateUnknown as Store;
+              interface OldStore extends Omit<Store, "actions"> {
+                gestures?: ActionData[];
+              }
+              const prevPersistedState = persistedStateUnknown as OldStore;
+              persistedState.actions = prevPersistedState.gestures ?? [];
+              delete prevPersistedState.gestures;
```
Comment (microbit-robert) [inline src/store.ts:855]: A slightly different way to write this so it's clear that we're always operating on the same object, but it's just as messy when it comes to type casting. Maybe there's a better options?

```suggestion
              interface OldStore extends Omit<Store, "actions"> {
                gestures?: ActionData[];
              }
              (persistedStateUnknown as Store).actions =
                (persistedStateUnknown as OldStore).gestures ?? [];
              delete (persistedStateUnknown as OldStore).gestures;
```

Comment (microbit-grace) [reply src/store.ts:855]: I agree that your suggestion is much more readable. Maybe @microbit-matt-hillsdon might have suggestions?

File: src/store.ts [hunk 2]
```diff
@@ -904,7 +886,7 @@ const getDataWindowFromActions = (actions: ActionData[]): DataWindow => {
 };
 
 // Get data window from actions on app load.
-const { gestures: actions } = useStore.getState();
+const { actions: actions } = useStore.getState();
```
Comment (microbit-matt-hillsdon) [inline src/store.ts:889]:
```suggestion
const { actions } = useStore.getState();
```

Comment (microbit-matt-hillsdon) [reply src/store.ts:889]: There are others, please search for "actions: actions".

Comment (microbit-grace) [reply src/store.ts]: [8f7306d](https://github.com/microbit-foundation/ml-trainer/pull/463/commits/8f7306d42de50c98a6b9a6b2d139ed1717cd7dc2)

File: src/store.ts (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply src/store.ts]: We tweaked this further after discussion. Resolving.

**Review by microbit-matt-hillsdon — APPROVED** [review]

**Pull request merged**